sonar-ballerina-plugin/src/main/java/io/ballerina/sonar/BallerinaSensor.java

```diff
@@ -0,0 +1,254 @@
+        Map<String, InputFile> pathAndInputFiles = new HashMap<>();
+        fileSystem.inputFiles(mainFilePredicate).forEach(inputFile -> {
+            pathAndInputFiles.put(Path.of(inputFile.uri()).toString(), inputFile);
+        });
+        sensorContext.config().get(ANALYSIS_RESULTS_FILE_PATH).ifPresentOrElse((analyzedResultsPath) ->
+                        processAnalyzedResultsReport(sensorContext, pathAndInputFiles, analyzedResultsPath),
+                () -> performLibraryCall(sensorContext, pathAndInputFiles)
+        );
+    }
+
+    public void processAnalyzedResultsReport(SensorContext context,
+                                             Map<String, InputFile> pathAndInputFiles,
+                                             String analyzedResultsFilePath) {
+        logger.info("Analyzing batch report: ", analyzedResultsFilePath);
+        String fileContent = getFileContent(analyzedResultsFilePath);
+        reportFileContent(context, pathAndInputFiles, fileContent);
+    }
+
+    public void performLibraryCall(SensorContext context, Map<String, InputFile> pathAndInputFiles) {
+        logger.info("Analyzing Ballerina project");
+        ProcessBuilder fileScan = new ProcessBuilder();
+        fileScan.directory(context.fileSystem().baseDir());
+        List<String> arguments = new ArrayList<>();
+        if (SystemUtils.IS_OS_WINDOWS) {
+            arguments.add("cmd");
+            arguments.add("/c");
+        } else {
+            arguments.add("sh");
+            arguments.add("-c");
+        }
+
+        arguments.add("bal scan --platform-triggered --platforms=sonarqube");
+        fileScan.command(arguments);
+        try {
+            fileScan.inheritIO();
+            Process process = fileScan.start();
+            int exitCode = process.waitFor();
+            if (exitCode == 0) {
+                String analyzedResultsFilePath = Paths.get(context.fileSystem().baseDir().getPath())
+                        .resolve(ISSUES_FILE_PATH)
+                        .toString();
+                String fileContent = getFileContent(analyzedResultsFilePath);
+                reportFileContent(context, pathAndInputFiles, fileContent);
+            } else {
+                logger.info("Unable to analyze ballerina file batch!");
+            }
+        } catch (IOException | InterruptedException ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+
+    private String getFileContent(String analyzedResultsFilePath) {
+        String fileContent = "";
+        try (
+                FileReader fileReader = new FileReader(analyzedResultsFilePath, StandardCharsets.UTF_8);
+                BufferedReader bufferedReader = new BufferedReader(fileReader)
+        ) {
+            fileContent = bufferedReader.lines().collect(Collectors.joining());
+        } catch (IOException ex) {
+            logger.info("Unable to retrieve analysis results!");
```

Comment:
Info? Also, is there a point returning empty string here and letting `reportFileContent` be called?

Reply:
Used `info` here and similar places. Also threw errors instead of returning empty strings. d42c5d59884f959feffce4a22dff4d00a3985afe